code/modules/station_goals/shield.dm

```diff
@@ -0,0 +1,171 @@
+//Station Shield
+// A chain of satellites encircles the station
+// Satellites be actived to generate a shield that will block unorganic matter from passing it.
+/datum/station_goal/station_shield
+	name = "Station Shield"
+
+/datum/station_goal/station_shield/get_report()
+	return {"The station is located in a zone full of space debris.
+			 We have a prototype shielding system you will deploy to reduce collision related accidents.
+
+			 You can order the satellites and control systems through cargo shuttle.
+			 "}
+
+
+/datum/station_goal/station_shield/on_report()
+	//Unlock 
+	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat]
+	P.special_enabled = TRUE
+
+	P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat_control]
+	P.special_enabled = TRUE
+
+/datum/station_goal/station_shield/check_completion()
+	if(..())
+		return TRUE
+	var/list/coverage = list()
+	for(var/obj/machinery/satellite/meteor_shield/A in machines)
+		if(!A.active || A.z != ZLEVEL_STATION)
+			continue
+		coverage |= view(A.kill_range,A)
+	if(coverage.len >= 400)
+		return TRUE
+	return FALSE
+
+
+/obj/item/weapon/circuitboard/machine/computer/sat_control
+	name = "circuit board (Satellite Network Control)"
+	build_path = /obj/machinery/computer/sat_control
+	origin_tech = "engineering=3"
+
+/obj/machinery/computer/sat_control
+	name = "Satellite control"
+	desc = "Used to control the satellite network."
+	circuit = /obj/item/weapon/circuitboard/machine/computer/sat_control
+	var/notice
+
+/obj/machinery/computer/sat_control/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
+										datum/tgui/master_ui = null, datum/ui_state/state = physical_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "sat_control", name, 400, 305, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/sat_control/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("toggle")
+			toggle(params["id"])
```

Comment:
This needs to be toggle(text2num(params["id"])) or do text2num(id) in the toggle proc below.
params returns a string